code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -430,114 +428,70 @@ components:
               enum:
                 - PRIVATEREPO
                 - PUBLICREPO
-              description: Artifact repository location. PUBLICREPO is used of public urls like github, helm repo etc, 
-                                  PRIVATEREPO  is used for private repo managed by the application developer.
+              description: Application repository and image URI information. PUBLICREPO is used of public urls like github, helm repo etc. PRIVATEREPO  is used for private repo managed by the application developer. Private repo can be accessed by using the app developer provided userName and password. Password is recommended to be the personal access token created by developer e.g. in Github repo. 
             url:
               $ref: '#/components/schemas/Uri'
             userName:
               type: string
-              description:  Username to acces the artifact repository
+              description:  Username to acces the Helm chart, docker-compose file or VM image repository
             password:
-              type: string      
-              description:  Password to acces the artifact repository
-            token:
-              type: string          
-              description:  Authorization token acces the artifact repository   
+              type: string
+              maxLength: 128
+              description:  Password or personal access token created by developerto acces the app repository
```

Comment:
It is important to have a checksum field so that when our platform pulls the image, we can verify that our copy is correct and we have pulled the exact image the user intended.

I suggest adding the field:
```
                checksum:
                  type: string
                  description: MD5 checksum for VM and file-based images, sha256 digest for containers
```

Reply:
Agree. Can incorporate this change.

Reply:
But there is one issue now. For VM the imagePath can point to the actual image file. But for the Helm charts or docker-compose file the name imagePath doesn't convey the intent. Also, as there could be multiple containers in the file there needs to be a list of container image name and associated checksums. Any thoughts?

Reply:
I agree Helm is tricky because a chart is really a directory with lots of files in it. There's no way to checksum it unless you package it into an archive. But for anything that's a single file (including docker-compose files), the intent is to checksum the file the OP will download. Anything that another program does, like docker parsing the docker-compose and downloading images, or helm parsing the chart files and downloading images, is really under the control of that program and out of our scope (we don't want to overlap with the functionality of docker or helm). So the intent is just to checksum whatever file is pointed to by imagePath. I would suggest that checksum be optional or not applicable for helm charts since it may be a directory structure instead of a single file.

Reply:
Changes done and committed to PR